class-components/src/components/Header.tsx

```diff
@@ -0,0 +1,76 @@
+import { ReactNode, useContext, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import ResourceSelector from './ResourceSelector';
+import InputBlock from './InputBlock';
+import {
+  selectPageNum,
+  selectSingleId,
+  selectText,
+  selectType,
+} from '../slices/headerSlice';
+import { ThemeContext } from './ThemeContext';
+import '../index.css';
+import ThemeSwitcher from './ThemeSwitcher';
+import { toggleAppError, toggleFetchError } from '../slices/errorSlice';
+
+function Header(): ReactNode {
+  const dispatch = useDispatch();
+  const typeFromStore = useSelector(selectType);
+  const textFromStore = useSelector(selectText);
+  const pageFromStore = useSelector(selectPageNum);
+  const singleIdFromStore = useSelector(selectSingleId);
+  const navigate = useNavigate();
+  const theme = useContext(ThemeContext);
+
+  const pageNum = pageFromStore > 1 ? `page=${pageFromStore}` : '';
```

Comment:
it's better to create helper, which will return an object with values that you need.